aasemble/django/apps/api/tests.py

```diff
@@ -306,3 +305,43 @@ def test_github_push_event(self, poll_one):
         from .tasks import github_push_event
         github_push_event("https://github.com/eric/project0")
         poll_one.delay.assert_called_with(1)
+
+class APIv2TagsTests(APIv1Tests):
+    list_url = '/api/v2/snapshots/'
+
+    def test_create_snapshot_tag(self):
+        data = {'mirrorset': 'http://testserver/api/v2/mirror_sets/60d0ba66-d343-404b-a6e6-5c141db11a54/',
+                'tags':[{'tag':'non-default'}]}
+        authenticate(self.client, 'eric')
+        response = self.client.post(self.list_url , data, format='json')
+        self.assertEquals(response.status_code, 201)
+        data['self']= response.data['self']
+        data['timestamp'] = response.data['timestamp']
+        self.assertEquals(data, response.data)
+        return response.data
+
+    def test_update_snapshot_tag(self):
+        data = {'tags':[{'tag':'non-default'}]}
+        authenticate(self.client, 'eric')
+        response = self.client.patch(self.list_url + '470688a8-7294-4c17-b020-1d67aebaf972/', data, format='json')
+        self.assertEquals(response.status_code, 200)
+        data['self']= response.data['self']
+        data['timestamp'] = response.data['timestamp']
+        data['mirrorset'] = response.data['mirrorset']
+        self.assertEquals(data, response.data)
+        return response.data
+
+    def test_update_snapshot_mirrorset_403(self):
```

Comment:
Why should this give a 403? I don't disagree, but from an outsider's perspective, it's not clear from the method name or a comment or anything else that what you're trying to do is change the snapshot's mirrorset attribute to something else than its original value (or so I'm guessing, I didn't actually check).

In other words: Make the test name more descriptive.